internal/ingress/controller/nginx.go

```diff
@@ -1022,3 +1004,62 @@ func cleanTempNginxCfg() error {
 
 	return nil
 }
+
+type redirect struct {
+	From    string
+	To      string
+	SSLCert ingress.SSLCert
+}
+
+func buildRedirects(servers []*ingress.Server) []*redirect {
+	names := sets.String{}
+	redirectServers := make([]*redirect, 0)
+
+	for _, srv := range servers {
+		if !srv.RedirectFromToWWW {
+			continue
+		}
+
+		var n string
+		if strings.HasPrefix(srv.Hostname, "www.") {
+			n = strings.TrimPrefix(srv.Hostname, "www.")
+		} else {
+			n = fmt.Sprintf("www.%v", srv.Hostname)
+		}
+
+		if names.Has(n) {
+			continue
+		}
+
+		klog.V(3).Infof("Creating redirect from %q to %q", srv.Hostname, n)
+		found := false
+		for _, esrv := range servers {
+			if esrv.Hostname == n {
+				found = true
+				break
+			}
+		}
+
+		if found {
+			continue
+		}
+
+		r := &redirect{
+			From: n,
+			To:   srv.Hostname,
+		}
```

Comment:
this is not consistent with https://github.com/kubernetes/ingress-nginx/pull/3637/files#diff-cde3fffe2425ad7efaa8add1d05ae2c0R1034. In the message it says redirect *from* `srv.Hostname` *to* `n`, but here you're reversing them.

Reply:
I think the log message need to be corrected, this construct itself is fine.

Reply:
done

Reply:
hmm I don't any change, it still says redirecting from hostname to n

Reply:
https://github.com/kubernetes/ingress-nginx/pull/3637/files#diff-cde3fffe2425ad7efaa8add1d05ae2c0R1056

Reply:
@aledbf now that log is wrong :D If you check my initial comment I'm pointing to line 1034 not 1056.

Reply:
ok, I renamed the variables to avoid confusions :wink: